sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -407,25 +407,7 @@ case class GpuSortArray(base: Expression, ascendingOrder: Expression)
 
   override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): cudf.ColumnVector = {
     val isDescending = isDescendingOrder(rhs)
-    val base = lhs.getBase()
-    withResource(base.getChildColumnView(0)) {child =>
-      withResource(child.copyToColumnVector()) {child =>
-        // When the average length of array is > 100
-        // and there are `-Nan`s in the data, the sort ordering
-        // of `Nan`s is inconsistent. So we need to normalize the data
-        // before sorting. This workaround can be removed after
-        // solving https://github.com/rapidsai/cudf/issues/11630
-        val normalizedChild = ColumnCastUtil.deepTransform(child) {
-          case (cv, _) if cv.getType == cudf.DType.FLOAT32 || cv.getType == cudf.DType.FLOAT64 =>
-              cv.normalizeNANsAndZeros()
-        }
-        withResource(normalizedChild) {normalizedChild =>
-          withResource(base.replaceListChild(normalizedChild)) {normalizedList =>
-            normalizedList.listSortRows(isDescending, true)
-          }
-        }
-      }
-    }
+    lhs.getBase.listSortRows(isDescending, true)
```

Comment:
Nit: This can also be combined with the line above for a shorter expression.

Reply:
The current code is more readable, I think.